### PR TITLE
Update TestIstioConfig

### PR DESCRIPTION
**Pull request by diemtvu**
The test was using hardcoded value for group and version. Change it to the one defined by ProtoSchema for the modernize IstioConfig.

### CODE REVIEW

File: pilot/test/mock/config.go [hunk 1]
```diff
@@ -415,33 +415,33 @@ func CheckIstioConfigTypes(store model.ConfigStore, namespace string, t *testing
...
 
 	for _, c := range cases {
 		if _, err := store.Create(model.Config{
 			ConfigMeta: model.ConfigMeta{
-				Type:      c.typ,
+				Type:      c.schema.Type,
 				Name:      name,
-				Group:     "config.istio.io",
-				Version:   "v1alpha2",
+				Group:     c.schema.Group + model.IstioAPIGroupDomain,
```
Comment (xiaolanz) [inline pilot/test/mock/config.go]: This should be ResourceGroup(c.schema).

Comment (diemtvu) [reply pilot/test/mock/config.go]: It would cause circular dependency. I guess we will need to refactor the code a bit later.

Comment (xiaolanz) [reply pilot/test/mock/config.go]: ok. ResourceGroup has to stay in crd.  Let's keep this for now.

**Pull request merged**